_posts/2016/2016-09-08-kubernetes-preparing-nodes.md

```diff
@@ -0,0 +1,259 @@
+EOF
+root@kubernetes:~$ apt-get update
+root@kubernetes:~$ apt-get install -y docker-engine
+root@kubernetes:~$ systemctl enable docker
+root@kubernetes:~$ systemctl stop docker
+root@kubernetes:~$ mkdir -p /etc/systemd/system/docker.service.d
+root@kubernetes:~$ cat > /etc/systemd/system/docker.service.d/docker.conf << "EOF"
+[Service]
+EnvironmentFile=/run/flanneld/docker_opts.env
+ExecStart=
+ExecStart=/usr/bin/docker daemon -H fd:// $DOCKER_OPTS $DOCKER_OPT_BIP $DOCKER_OPT_MTU $DOCKER_OPT_IPMASQ
+EOF
+root@kubernetes:~$ systemctl daemon-reload
+root@kubernetes:~$ systemctl enable docker
+root@kubernetes:~$ systemctl restart docker
+{% endhighlight %}
+
+*drop-in* overwrites the start command.
+The new one which is loaded via the option *EnvironmentFile*.
+It contains the variables from the file _/run/flanneld/docker_opts.env_ that was created by *flanneld*.
+
+# Kubelet
+
+[Kubelet](http://kubernetes.io/docs/admin/kubelet/) controls the directory _/etc/kubernetes/manifests_ and creates Docker containers from the Pod definitions located here.
+Therefore it also needs root rights.
+Please be aware of the fact that the Kubelet version has to be compatible to the one of the API server.
+
+### Installation
+Kubelet is part of the official Kubernetes release.
+It is simply loaded to every Kubernetes node and unpacked there.
+
+{% highlight bash %}
+root@kubernetes:/tmp$ curl -L -O https://github.com/kubernetes/kubernetes/releases/download/v1.2.2/kubernetes.tar.gz
+root@kubernetes:/tmp$ tar -xzf kubernetes.tar.gz
+root@kubernetes:/tmp$ cd kubernetes/server
+root@kubernetes:/server$ tar -xzf kubernetes-server-linux-amd64.tar.gz
+root@kubernetes:/server$ cp kubernetes/server/bin/kubelet /usr/local/bin/kubelet
+root@kubernetes:/server$ cd /tmp
+root@kubernetes:/tmp$ rm -rf kubernetes
+root@kubernetes:/tmp$ chmod 755 /usr/local/bin/kubelet
+root@kubernetes:/tmp$ mkdir -p /etc/kubernetes/manifests
+{% endhighlight %}
+
+In addition we have to add [Kernel boot parameter](https://github.com/kubernetes/kubernetes/issues/9837).
+The example how we do this using *sed*.
+
+{% highlight bash %}
+root@kubernetes:~$ sed -i "s/^GRUB_CMDLINE_LINUX_DEFAULT=\"\(.*\)\"$/GRUB_CMDLINE_LINUX_DEFAULT=\"\1\ cgroup_enable=memory\"/g" /etc/default/grub
+root@kubernetes:~$ update-grub
+root@kubernetes:~$ reboot
+{% endhighlight %}
+
+## Summary
+
+The external __Service node__ is used as a load balancer.
+Working on the [Transport layer](https://en.wikipedia.org/wiki/Transport_layer) of the [OSI reference model](https://en.wikipedia.org/wiki/OSI_model) it does not terminate SSL but simply routes the traffic.
+
+On all __Kubernetes nodes__ run various services.
+*Flanneld* is required to run the *Docker engine*.
+It stores data for the networks in the *Etcd cluster* and therefore needs a client certificate.
+The *Docker engine* is a standard installation.
+It uses its own *drop-in* to be able to handle the parameters provided by *flanneld*.
```

Comment:
Add this chapter:

## Related post
* [How to set up a HA Kubernetes cluster: etcd cluster with SSL](https://developer.epages.com/blog/2016/08/09/kubernetes-etcd-cluster.html)